File: C/main_lin.c
#include <math.h>
#include <stdio.h>
#include <stdlib.h>

double pofk_enhancement_linear(double _z, double _fR0, double _k);

int main(int argc, char **argv){
  if(argc < 3){
    printf("Run as ./test fofr0 redshift\n");
    exit(1);
  }
  double fofr0 = atof(argv[1]);
  double z    = atof(argv[2]);

  int nk = 30;
  double kmin = 0.01;
  double kmax = 10.0;
  printf("# For fR0 = %e and z = %f we have the following (k, ratio(k)): \n",fofr0,z);
  for(int i = 0; i < nk; i++){
    double k = exp(log(kmin) + log(kmax/kmin)*i/(double)(nk-1));
    printf("%lf    %lf\n", k, pofk_enhancement_linear(z, fofr0, k));
  }
}

File: Python/main_lin.py
from fit_lin import pofk_enhancement_linear
import numpy as np

# fR0 and redshift
fofr0 = 1e-5
z = 0.0

# k-range to output 
nk = 30
kmin = 0.01
kmax = 10.0

print "For fR0 = ", fofr0, " and z = ", z, " we have the following (k, ratio(k)):"
for i in range(nk):
  k = np.exp(np.log(kmin) + np.log(kmax/kmin) * i /(nk-1.0))
  print k,  pofk_enhancement_linear(z, fofr0, k)


print pofk_enhancement_linear(z, fofr0, 1.0)
